code/modules/lighting/lighting_corner.dm

```diff
@@ -0,0 +1,97 @@
+/var/list/datum/lighting_corner/all_lighting_corners = list()
+// Because we can control each corner of every lighting overlay.
+// And corners get shared between multiple turfs (unless you're on the corners of the map, then 1 corner doesn't).
+// For the record: these should never ever ever be deleted, even if the turf doesn't have dynamic lighting.
+
+// This list is what the code that assigns corners listens to, the order in this list is the order in which corners are added to the /turf/corners list.
+/var/list/LIGHTING_CORNER_DIAGONAL = list(NORTHEAST, SOUTHEAST, SOUTHWEST, NORTHWEST)
+
+/datum/lighting_corner
+	var/list/turf/masters                 = list()
+	var/list/datum/light_source/affecting = list() // Light sources affecting us.
+	var/active                            = FALSE  // TRUE if one of our masters has dynamic lighting.
+
+	var/x     = 0
+	var/y     = 0
+	var/z     = 0
+
+	var/lum_r = 0
+	var/lum_g = 0
+	var/lum_b = 0
+
+	var/needs_update = FALSE
+
+/datum/lighting_corner/New(var/turf/new_turf, var/diagonal)
+	. = ..()
+
+	all_lighting_corners += src
+
+	masters[new_turf] = turn(diagonal, 180)
+	z = new_turf.z
+
+	var/vertical   = diagonal & ~(diagonal - 1) // The horizontal directions (4 and 8) are bigger than the vertical ones (1 and 2), so we can reliably say the lsb is the horizontal direction.
+	var/horizontal = diagonal & ~vertical       // Now that we know the horizontal one we can get the vertical one.
+
+	x = new_turf.x + (horizontal == EAST  ? 0.5 : -0.5)
+	y = new_turf.y + (vertical   == NORTH ? 0.5 : -0.5)
+
+	// My initial plan was to make this loop through a list of all the dirs (horizontal, vertical, diagonal).
+	// Issue being that the only way I could think of doing it was very messy, slow and honestly overengineered.
+	// So we'll have this hardcode instead.
+	var/turf/T
+	var/i
+
+	// Diagonal one is easy.
+	T = get_step(new_turf, diagonal)
+	if(T) // In case we're on the map's border.
+		masters[T]   = diagonal
+		i            = LIGHTING_CORNER_DIAGONAL.Find(turn(diagonal, 180))
+		T.corners[i] = src
+
+	// Now the horizontal one.
+	T = get_step(new_turf, horizontal)
+	if(T) // Ditto.
+		masters[T]   = ((T.x > x) ? EAST : WEST) | ((T.y > y) ? NORTH : SOUTH) // Get the dir based on coordinates.
+		i            = LIGHTING_CORNER_DIAGONAL.Find(turn(masters[T], 180))
+		T.corners[i] = src
+
+	// And finally the vertical one.
+	T = get_step(new_turf, vertical)
+	if(T)
+		masters[T]   = ((T.x > x) ? EAST : WEST) | ((T.y > y) ? NORTH : SOUTH) // Get the dir based on coordinates.
+		i            = LIGHTING_CORNER_DIAGONAL.Find(turn(masters[T], 180))
+		T.corners[i] = src
+
+	spawn() // Lighting overlays get initialized AFTER corners, so this spawn() will make sure the activity (which checks for overlays) is updated after the overlays are generated.
```

Comment:
During startup (with away missions disabled), over 69,000 corners get created, and every single one spawns right here. Spawning is embarrassingly expensive - on my machine, `lighting_corner/New` accounts for 42 seconds of CPU time used, drastically increasing startup time.

I tried replacing these two lines of code with the following...
```
var/global/list/lighting_corner_update_list
if(!lighting_corner_update_list)
	lighting_corner_update_list = list()
	lighting_corner_update_list += src
	spawn(-1)
		src = null
		sleep(0)
		for(var/datum/lighting_corner/corner in lighting_corner_update_list)
			corner.update_active()
		log_to_dd("Updated [lighting_corner_update_list.len] corners.")
		lighting_corner_update_list = null
else
	lighting_corner_update_list += src
```
... and `lighting_corner/New` updated all 69166 corners with a single spawn, dropping to 0.6 seconds of CPU time. Loading Underground Outpost 45 adds another 19072 corners, bringing it to 0.8 seconds. This seems to pretty much completely eliminate the extra startup time - and probably whatever lag is caused by regenerating shuttle corners - without any negative consequences.

You may want to consider doing something like this to improve startup times on VG, @PJB3005.